roles/edpm_fips/tasks/enable.yml

```diff
@@ -0,0 +1,75 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Enable FIPS
+  become: true
+  ansible.builtin.command: fips-mode-setup --enable
+  register: _fms_enable
+  changed_when: _fms_enable.rc == 0
+
+- name: Ensure that `fips=1` flag is present in GRUB parameter file
```

Comment:
we shoudl not modify the files directly, i think in stead we should instead delegate this to the edpm_kernel role.

Reply:
Actually you should confirm if fips-mode-setup --enable manages the modification of kernel arguments itself. Certainly that is all that is required in the diskimage-builder element: https://opendev.org/openstack/diskimage-builder/src/branch/master/diskimage_builder/elements/fips/pre-install.d/01-setup-fips#L30

Reply:
Yeah I know my original role https://opendev.org/zuul/zuul-jobs/src/branch/master/roles/enable-fips/tasks/rhel.yaml modified the files, but things may have changed since then.   So definitely check to see if fips-mode-setup enable does whatever you need.

Reply:
i read the code of [fips-mode-setup](https://gitlab.com/redhat-crypto/fedora-crypto-policies/-/raw/rhel9/fips-mode-setup), it's a bash script so it quite easy. It works as long `grubby` is installed, otherwise you need to patch the files manually. IMO i'd add `grubby` as required package dependency and let the script handle `fips=` flag in the boot loader. WDYT @SeanMooney @steveb @vakwetu ?

Reply:
Let's conditionally add the `grubby` dependency based on whether or not the user wants to use FIPS. But otherwise, I think it would be the easiest option to just add `grubby`.